extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcIdentityProvider.java

```diff
@@ -169,9 +169,16 @@
                             builder.addCredential(tokenCred);
                             OidcUtils.setSecurityIdentityUserInfo(builder, userInfo);
                             OidcUtils.setSecurityIdentityConfigMetadata(builder, resolvedContext);
+                            String principalMember;
                             if (result.introspectionResult.containsKey(OidcConstants.INTROSPECTION_TOKEN_USERNAME)) {
+                                principalMember = OidcConstants.INTROSPECTION_TOKEN_USERNAME;
+                            } else if (result.introspectionResult.containsKey(OidcConstants.INTROSPECTION_TOKEN_SUB)) {
+                                // fallback to "sub", if "username" is not present
+                                principalMember = OidcConstants.INTROSPECTION_TOKEN_SUB;
+                            }
+                            if (!principalMember.isEmpty()) {
```

Comment:
@KarstenWintermann Please remove this condition - this will avoid all the side-effects in case some other provider sets `preferredUserName` for example, etc